camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/ServiceFailureDetectorYamlTest.java

```diff
@@ -0,0 +1,189 @@
+    
+    static final String catalogYamlSimple = Joiner.on("\n").join(
+            "brooklyn.catalog:",
+            "  id: " + appId,
+            "  version: " + appVersion,
+            "  itemType: entity",
+            "  item:",
+            "    type: " + TestEntity.class.getName(),
+            "    brooklyn.enrichers:",
+            "    - type: " + ServiceFailureDetector.class.getName());
+
+    static final String catalogYamlWithDsl = Joiner.on("\n").join(
+            "brooklyn.catalog:",
+            "  id: my-app",
+            "  version: 1.2.3",
+            "  itemType: entity",
+            "  item:",
+            "    services:",
+            "    - type: " + TestEntity.class.getName(),//FailingEntity.class.getName(),
+            "      brooklyn.parameters:",
+            "      - name: custom.stabilizationDelay",
+            "        type: " + Duration.class.getName(),
+            "        default: 1ms",
+            "      - name: custom.republishTime",
+            "        type: " + Duration.class.getName(),
+            "        default: 1m",
+            "      brooklyn.enrichers:",
+            "      - type: " + ServiceFailureDetector.class.getName(),
+            "        brooklyn.config:",
+            "          serviceOnFire.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityFailed.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityRecovered.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityFailed.republishTime: $brooklyn:config(\"custom.republishTime\")");
+
+    static final String catalogYamlWithDslReferenceParentDefault = Joiner.on("\n").join(
+            "brooklyn.catalog:",
+            "  id: my-app",
+            "  version: 1.2.3",
+            "  itemType: entity",
+            "  item:",
+            "    brooklyn.parameters:",
+            "    - name: custom.stabilizationDelay",
+            "      type: " + Duration.class.getName(),
+            "      default: 1ms",
+            "    - name: custom.republishTime",
+            "      type: " + Duration.class.getName(),
+            "      default: 1m",
+            "    services:",
+            "    - type: " + TestEntity.class.getName(),//FailingEntity.class.getName(),
+            "      brooklyn.enrichers:",
+            "      - type: " + ServiceFailureDetector.class.getName(),
+            "        brooklyn.config:",
+            "          serviceOnFire.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityFailed.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityRecovered.stabilizationDelay: $brooklyn:config(\"custom.stabilizationDelay\")",
+            "          entityFailed.republishTime: $brooklyn:config(\"custom.republishTime\")");
```

Comment:
You're right. I change it to have two children, and the test failed (because it didn't merge to the top-level). I've therefore also changed the yaml to use `$brooklyn:scopeRoot().config(...)`.